### ISSUE TITLE
Delete rotated logs

### ISSUE THREAD

**Issue by cristim** [rogerc/file-stream-rotator#21]
It should only keep the last X items, where X should have a sane default and should be configurable by the user.

**Comment by jrpomeroy** [comment]
A log file retention policy is something I really wish this had.  I'm using it for now but still looking for alternatives.

**Comment by rogerc** [comment]
The goal for this module is to be as light as possible while performing the core function of rotating logs effectively. Log retention policy can be quite complex and resource heavy depending on what is required (compression, parsing, pushing to AWS/other,etc). For now, to support this I've added the on.rotate and on.new events to handle log management as required.

For log management, it's best to do it in the quiet periods and by a scheduler to guarantee that it will happen. I'd rather not add setInterval or any other "background" task that could affect the performance of the main application.

**Comment by lulzmachine** [comment]
I'm really looking for this too. Just a feature to say "maximum 10 log files" would be enough.

**Comment by rogerc** [comment]
I've added a clean-up option. Not published yet. To avoid doing a directory listing on a path containing thousands of files, it keeps track of the files it has created and will only delete the files in the audit log. This also reduces the chances of deleting the wrong files. I will publish it on tomorrow/Monday

**Comment by rogerc** [comment]
Examples below as found in the README.

```javascript
    // Rotate on the hour or any specified number of hours and keep 10 files
    var rotatingLogStream = require('file-stream-rotator').getStream({filename:"/tmp/test.log", frequency:"1h", verbose: false, max_logs: 10});

    // Rotate on the hour or any specified number of hours and keep 10 days
    var rotatingLogStream = require('file-stream-rotator').getStream({filename:"/tmp/test.log", frequency:"1h", verbose: false, max_logs: "10d"});

    // Rotate on the hour or any specified number of hours and keep 10 days and store the audit file in /tmp/log-audit.json
    var rotatingLogStream = require('file-stream-rotator').getStream({filename:"/tmp/test.log", frequency:"1h", verbose: false, max_logs: "10d", audit_file: "/tmp/log-audit.json"});
```

**Comment by lulzmachine** [comment]
That sounds cool!
When you say "it keepst rack of the files it has created", do you mean in memory or between restarts as well?

**Comment by rogerc** [comment]
They get written to a file with a default location of the root folder of the static part of where the logs are written. By that I mean that if the path contains a %DATE% placeholder, it will be in the directory preceding that, e.g. /tmp/logs/%DATE%/server.log, it will store the audit file in /tmp/logs/.audit.json. The name and path can be changed through configuration. 

By writing them to a file, as you mention, it will handle application restarts/crashes and continue where it left off.

The implementation I think should have little impact on performance of the host application as unlinking a defined file should be almost instant in most OS.

**Comment by rogerc** [comment]
version with log deletion option has been published to npm - file-stream-rotator@0.2.0

https://www.npmjs.com/package/file-stream-rotator
